swxsoc/util/config.py

```diff
@@ -28,202 +27,149 @@
-    """
-    config = configparser.RawConfigParser()
-
-    # Get locations of configuration files to be loaded
-    config_files = _find_config_files()
+    If a configuration file does not exist in the user's home directory,
+    it will read in the defaults from the package's data directory.
 
-    # Read in configuration files
-    config.read(config_files)
-
-    # This is to fix issue with AppDirs not writing to /tmp/ in AWS Lambda
-    if os.getenv("LAMBDA_ENVIRONMENT"):
-        config.set("logger", "log_to_file", "False")
-
-    # Specify the working directory as a default so that the user's home
-    # directory can be located in an OS-independent manner
-    if not config.has_option("general", "working_dir"):
-        config.set("general", "working_dir", str(Path.home() / "hermes"))
-
-    # Set the download_dir to be relative to the working_dir
-    working_dir = Path(config.get("general", "working_dir"))
-    download_dir = Path(config.get("downloads", "download_dir"))
-    config.set(
-        "downloads",
-        "download_dir",
-        (working_dir / download_dir).expanduser().resolve().as_posix(),
-    )
+    The selected mission can be overridden by setting the `SWXSOC_MISSION`
+    environment variable. This environment variable will take precedence
+    over the mission specified in the configuration file.
 
-    # Set Mission-Specific Configuration
-    config.set("mission", "inst_names", json.loads(config.get("mission", "inst_names")))
-    config.set(
-        "mission",
-        "inst_shortnames",
-        json.loads(config.get("mission", "inst_shortnames")),
-    )
-    config.set(
-        "mission", "inst_fullnames", json.loads(config.get("mission", "inst_fullnames"))
-    )
-    config.set(
-        "mission",
-        "inst_targetnames",
-        json.loads(config.get("mission", "inst_targetnames")),
-    )
-    config.set(
-        "mission",
-        "inst_to_shortname",
-        dict(
-            zip(
-                config.get("mission", "inst_names"),
-                config.get("mission", "inst_shortnames"),
-            )
-        ),
-    )
-    config.set(
-        "mission",
-        "inst_to_targetname",
-        dict(
-            zip(
-                config.get("mission", "inst_names"),
-                config.get("mission", "inst_targetnames"),
-            )
-        ),
-    )
-    config.set(
-        "mission",
-        "inst_to_fullname",
-        dict(
-            zip(
-                config.get("mission", "inst_names"),
-                config.get("mission", "inst_fullnames"),
-            )
+    Returns:
+        dict: The loaded configuration data.
+    """
+    config_path = Path(_get_user_configdir()) / "config.yml"
+    if not config_path.exists():
+        config_path = Path(swxsoc.__file__).parent / "data" / "config.yml"
+
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    selected_mission = os.getenv("SWXSOC_MISSION", config["selected_mission"])
```

Comment:
Do you think we should describe this environment variable and how to use it inside the customization docs?